### PR TITLE
[docs] Update docs for impossible TzDate casts

### PR DESCRIPTION
**Pull request by igormunkin**
Follows up #5805

### Changelog category

* Documentation (changelog entry is not required)

### CODE REVIEW

File: ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md [hunk 1]
```diff
@@ -460,6 +460,23 @@ select
 	DateTime::Format("%Y-%m-%d %Z")(AddTimezone($x, "Europe/Moscow")), -- 2019-10-22 Europe/Moscow
 ```
 
+It's worth mentioning, that several TzDatetime or TzTimestamp values with the positive timezone offset, cannot be casted to TzDate. Consider the example below:
```
Comment (blinkov) [inline ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md:463]:
```suggestion
It's worth mentioning that several `TzDatetime` or `TzTimestamp` values with a positive timezone offset cannot be cast to `TzDate`. Consider the example below:
```

File: ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md [hunk 2]
```diff
@@ -460,6 +460,23 @@ select
...
+
+```yql
+select cast(TzDatetime("1970-01-01T23:59:59,Europe/Moscow") as TzDate)
```
Comment (blinkov) [inline ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md:466]:
```suggestion
SELECT CAST(TzDatetime("1970-01-01T23:59:59,Europe/Moscow") as TzDate);
```

File: ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md [hunk 3]
```diff
@@ -460,6 +460,23 @@ select
...
+/* Fatal: Timestamp 1970-01-01T23:59:59.000000,Europe/Moscow cannot be casted to TzDate */
+```
+
+There is no valid value, starting from the Unix epoch, to represent the 01/01/1970 midnight for Europe/Moscow timezone. As a result, such cast is impossible and fails at runtime.
+
+At the same time, values with the negative timezone offset are converted the right way:
```
Comment (blinkov) [inline ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md:472]:
```suggestion
Starting from the Unix epoch, there is no valid value representing midnight on 01/01/1970 for the Europe/Moscow timezone. As a result, such a cast is impossible and fails at runtime.

At the same time, values with a negative timezone offset are converted correctly:
```

Comment (blinkov) [inline ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md:472]: It sounds like this issue doesn't exist for negative timezone offsets, but in reality, it probably still exists around the maximum possible date?

File: ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md [hunk 4]
```diff
@@ -460,6 +460,23 @@ select
...
+
+```yql
+select cast(TzDatetime("1970-01-01T23:59:59,America/Los_Angeles") as TzDate)
```
Comment (blinkov) [inline ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md:475]:
```suggestion
SELECT CAST(TzDatetime("1970-01-01T23:59:59,America/Los_Angeles") as TzDate);
```

**Review by blinkov — DISMISSED** [review]

File: ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md (hunk 3 above)
Comment (igormunkin) [reply ydb/docs/en/core/yql/reference/yql-core/udf/list/datetime.md:472]: First of all, there is no issue with negative timezone offsets.

There is also no way to face the issue with the positive timezone offsets. The datetime value, representing `UINT32_MAX` value, is `2106-02-07T11:28:15Z`. The maximum year value available in YQL is 2106 (non-inclusive), that means the maximum datetime value is `2105-12-31T23:59:59` for every timezone available.

However, 2106-01-01 might be returned for the aforementioned timezones. Consider the following:
```
SELECT CAST(AddTimezone(Datetime("2105-12-31T23:59:59Z"), "Europe/Moscow") as TzDate);
/* 2106-01-01,Europe/Moscow */
```

This value can be represented by UINT32 type, so no issue occurs.

File: ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md [hunk 5]
```diff
@@ -464,6 +464,23 @@ select
 	DateTime::Format("%Y-%m-%d %Z")(AddTimezone($x, "Europe/Moscow")), -- 2019-10-22 Europe/Moscow
 ```
 
+Стоит отметить, что значения в типaх TzDatetime и TzTimestamp, имеющие положительное смещение таймзоны, не могут быть преобразованы в тип TzDate. Рассмотрим следующий пример:
+
+```yql
+select cast(TzDatetime("1970-01-01T23:59:59,Europe/Moscow") as TzDate)
```
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md:470]: SELECT CAST(TzDatetime("1970-01-01T23:59:59,Europe/Moscow") as TzDate);
```

File: ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md [hunk 6]
```diff
@@ -464,6 +464,23 @@ select
...
+/* Fatal: Timestamp 1970-01-01T23:59:59.000000,Europe/Moscow cannot be casted to TzDate */
+```
+
+Не существует такого значения, начиная с Unix epoch, которым можно представить полночь 01.01.1970 для московского времени. В результате, такое преобразование считается невозможным и генерирует ошибку времени выполнения.
+
+В то же время, значения, имеющие отрицательное смещение таймзоны, возвращают корректный результат:
+
+```yql
+select cast(TzDatetime("1970-01-01T23:59:59,America/Los_Angeles") as TzDate)
```
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md:479]: SELECT CAST(TzDatetime("1970-01-01T23:59:59,America/Los_Angeles") as TzDate);
```

File: ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md (hunk 5 above)
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md:470]:
```suggestion
SELECT CAST(TzDatetime("1970-01-01T23:59:59,Europe/Moscow") as TzDate);
```

File: ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md (hunk 6 above)
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md:479]:
```suggestion
SELECT CAST(TzDatetime("1970-01-01T23:59:59,America/Los_Angeles") as TzDate);
```

File: ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md [hunk 7]
```diff
@@ -464,6 +464,23 @@ select
 	DateTime::Format("%Y-%m-%d %Z")(AddTimezone($x, "Europe/Moscow")), -- 2019-10-22 Europe/Moscow
 ```
 
+Стоит отметить, что значения в типaх TzDatetime и TzTimestamp, имеющие положительное смещение таймзоны, не могут быть преобразованы в тип TzDate. Рассмотрим следующий пример:
```
Comment (vitstn) [inline ydb/docs/ru/core/yql/reference/yql-core/udf/list/datetime.md:467]: значения -> некоторые значения

**Review by fomichev3000 — APPROVED** [review]

**Pull request merged**